programs/mango-v4/src/state/bank.rs

```diff
@@ -528,19 +548,25 @@ impl Bank {
         position: &mut TokenPosition,
         already_borrowed_native_amount: I80F48,
         now_ts: u64,
-    ) -> Result<(bool, I80F48)> {
+    ) -> Result<WithdrawResult> {
         let loan_origination_fee = self.loan_origination_fee_rate * already_borrowed_native_amount;
         self.collected_fees_native += loan_origination_fee;
 
-        let (position_is_active, _) = self.withdraw_internal_wrapper(
-            position,
+        let position_is_active = self
+            .withdraw_internal_wrapper(
+                position,
+                loan_origination_fee,
+                false,
+                !position.is_in_use(),
+                now_ts,
+            )?
+            .position_is_active;
+
+        Ok(WithdrawResult {
+            position_is_active,
             loan_origination_fee,
-            false,
-            !position.is_in_use(),
-            now_ts,
-        )?;
-
-        Ok((position_is_active, loan_origination_fee))
+            loan_amount: already_borrowed_native_amount,
```

Comment:
Hmm, this is a tricky case. The borrow has already happened before and withdraw has returned a WithdrawResult with loan_origination_fee == 0 and loan_amount > 0. To be consistent, we should return loan_amount = 0 here.

Otherwise there'd be double-counting of loans!